Rethink tooltips for search options

flodolo opened the pull request:
A few thoughts on the current implementation (tooltip on hover):
* It's annoying if you already know what the option does.
* It's broken on touch devices or, at least, on iOS. Spent an entire afternoon trying to figure out how to deal with it, but the series of events fired is completely inconsistent.

One possible alternative is, instead of showing a tooltip, opening a help page somewhere with more general instructions on how to use Transvision.

TheoChevalier commented on app/views/search_form.php:
```diff
@@ -146,8 +157,7 @@ class="search_options"
                         />
-                        <label for="each_word" aria-describedby="tooltip_eachword">Each Word</label>
-                        <div id="tooltip_eachword" class="search_tooltip" role="tooltip">If selected, search for <strong>each word</strong> in the search query (at least 2 characters long). By default, the search query is used as a whole.</div>
+                        <label>Each Word</label>
```
Should be `<label for="each_word">Each Word</label>` (can’t be checked otherwise)

flodolo replied:
Gah, removed too much 😒

TheoChevalier commented on app/views/search_form.php:
```diff
@@ -157,8 +167,7 @@ class="search_options"
                         />
-                        <label for="entire_string" aria-describedby="tooltip_entirestring">Entire String</label>
-                        <div id="tooltip_entirestring" class="search_tooltip" role="tooltip">If selected, the <strong>entire string</strong> needs to match the search query. By default, partial matches are included in the results.</div>
+                        <label>Entire String</label>
```
Should be `<label for="entire_string">Each Word</label>`

TheoChevalier commented on app/views/search_form.php:
```diff
@@ -168,8 +177,7 @@ class="search_options"
                         />
-                        <label for="entire_words" aria-describedby="tooltip_entirewords">Entire Words</label>
-                        <div id="tooltip_entirewords" class="search_tooltip" role="tooltip">If selected, each search term needs to match an <strong>entire word</strong>. By default, partial word matches are included in the results.</div>
+                        <label>Entire Words</label>
```
Should be `<label for="entire_words">Entire Words</label>`

TheoChevalier commented on app/views/search_form.php:
```diff
@@ -127,6 +127,18 @@ class="mainsearch_default_checkbox"
                     <label>Advanced Search options</label>
+                    <a href="#" id="tooltip_search"><span>i</span></a>
+                    <div class="tooltip" id="tooltip_search_text">
+                        <a href="#" id="tooltip_search_close" data-tooltip-id="tooltip_search_text" title="Close this panel">x</a>
```
nit: “×” character (U+00D7) vs. x (U+0078) for consistency with the clear button in the search field. If we’re switching characters, we will need to increase font-size

TheoChevalier commented on web/style/transvision.css:
```diff
@@ -782,6 +790,83 @@ input[type="checkbox"]:disabled + label {
+}
+
+#tooltip_search_close:hover {
+    border: 1px solid #ccc;
```
Do we really need this? It feels a bit weird as it’s moving the button by 1px on hover. If we really want a border to identify the click area, maybe we should use a transparent border on the default state.

TheoChevalier commented on web/style/transvision.css:
```diff
@@ -782,6 +790,83 @@ input[type="checkbox"]:disabled + label {
+    font-family: 'Fira Sans UltraLight', Arial, sans-serif;
+    font-size: 25px;
+    line-height: 0.5em;
+    color: #fff;
```
I think we want to add `text-decoration: none;` here to remove the underline from the link on hover

TheoChevalier commented:
I like this direction better than the current master (been annoyed by the tooltips too after a few days of usage). And I also think the always visible “i” icon won’t hurt for this critical area

flodolo commented:
I think I've addressed all the issues, thanks for spotting them.

I removed the hover on the x, and slightly enlarged/moved it. Not sure why I didn't notice the text-decoration on hover :-\

flodolo commented:
Let me check that X again, it still doesn't look right.

flodolo commented:
> Let me check that X again, it still doesn't look right.

Ehm, never mind. VM didn't pull the latest changes…
